typesafe/shared/src/main/scala/zio/config/typesafe/TypesafeConfigProvider.scala

```diff
@@ -59,7 +58,8 @@ object TypesafeConfigProvider {
 
     ConfigProvider.fromMap(
       indexedMapWithHiddenDelimiter,
-      pathDelim = hiddenDelim
+      pathDelim = hiddenDelim,
+      seqDelim = if(enableCommaSeparatedValueAsList) "," else hiddenDelim
```

Comment:
Would it be better if the user can provide their own separator like `|` ? For example `valueSeparator: Option[String]` this line can be like this:

```scala
seqDelim = valueSeparator.getOrElse(hiddenDelim)
```

I think it's not worth it for HOCON, but for other providers comma might not be the ideal separator.